packages/base/src/TabView/TabView.tsx

```diff
@@ -56,33 +56,41 @@ export const TabViewBase: RneFunctionComponent<TabViewBaseProps> = ({
 }) => {
   const { current: translateX } = React.useRef(new Animated.Value(0));
   const currentIndex = React.useRef(value);
-  const length = React.Children.count(children);
+  const validChildren = React.Children.toArray(children);
+  const length = validChildren.length;
   const window = useWindowDimensions();
 
-  const onPanResponderRelease = (
-    _: GestureResponderEvent,
-    { dx, dy }: PanResponderGestureState
-  ) => {
-    if (
-      (dx > 0 && currentIndex.current <= 0) ||
-      (dx < 0 && currentIndex.current >= length - 1)
-    ) {
-      return;
+  useEffect(() => {
+    if (currentIndex.current > length - 1) {
+      onChange(length - 1);
     }
-    if (Math.abs(dy) > Math.abs(dx)) {
-      return;
-    }
-    const position = dx / -window.width;
-    const next = position > value ? Math.ceil(position) : Math.floor(position);
-    onChange?.(currentIndex.current + next);
```

Comment:
For some reason this logic broke with conditional children. I fixed by simply adding or subtracting 1 from the index based on the sign of dx

Reply:
After a second look, the original code was setting "value" only once upon initial render when panResponder was initialized with useRef. Since value was, by default, set to 0 - this code was effectively doing the same thing as my fix. This code could have caused bugs in its current state - for example, if the developer mounted the component with a starting value for "value" of a number greater than 0, it could have prevented the index from being incremented since position would never be greater than value. The reason it broke with dynamic children was because "value" was permitted to update whenever the dependency on length was added. 

Would it be worthwhile to consider added a minimum dx threshold to change TabViews? Right now, as long as Abs(dx) is greater than Abs(dy), even if it's a very small number, a pan will always initiate the view change. Is that desirable? Or should we be checking if the pan is more than, say, 25% of the screen width?

Reply:
I am making changes here https://github.com/react-native-elements/react-native-elements/pull/3380

Reply:
Looks great - exactly what I was thinking of. It looks like to implement my fix in your new code, `React.Children.count(children)` on line 76 will need changed to `React.Children.toArray(children).length`, and that same value will need to be included in the dependency array on 101 as well. 

Nice work.

Reply:
Hey, no issue, would merge your PR first